### PR TITLE
feat: add ExpandTabFunc for custom tab expansion

### LINKED ISSUE #1: Add ExpandTabFunc for custom tab expansion
## Proposal

Add a new method `ExpandTabFunc` to `Condition` that generalizes `ExpandTab` with a callback for custom tab replacement:

```go
func (c *Condition) ExpandTabFunc(s string, fn func(nSpaces int) string) string
```

The callback receives the number of spaces that a tab would normally expand to (based on the current column and tab width), and returns the replacement string. This enables use cases where tabs need to be replaced with something other than plain spaces while maintaining correct column alignment.

## Motivation

In [spanner-mycli](https://github.com/apstndb/spanner-mycli), tab characters in Spanner column values need to be visualized with a visible marker (e.g., `→` plus padding spaces) so users can distinguish tabs from spaces in table output. The current `ExpandTab` always replaces tabs with spaces, so spanner-mycli must manually split on `\t`, track column positions with `StringWidth`, and compute tab stops — essentially reimplementing the core of `ExpandTab`.

See: https://github.com/apstndb/spanner-mycli/issues/568

## Example usage

```go
// Replace tabs with "→" + (n-1) padding spaces
cond := &tabwrap.Condition{TabWidth: 4}
result := cond.ExpandTabFunc("abc\tdef", func(nSpaces int) string {
    return "→" + strings.Repeat(" ", nSpaces-1)
})
// result: "abc→def"
```

## Current workaround

spanner-mycli currently uses a `strings.Split` + `StringWidth` approach to manually track column positions and compute tab-stop spacing. `ExpandTabFunc` would replace this with a single library call.

### PR DESCRIPTION
**Pull request by apstndb**
## Summary

- Add `ExpandTabFunc` method to `Condition` that generalizes `ExpandTab` with a callback `func(nSpaces int) string`
- The callback receives the number of spaces a tab would normally expand to, enabling custom tab replacement (e.g., visible markers like `→` + padding)
- Refactor `ExpandTab` to delegate to `ExpandTabFunc`, eliminating duplication

Closes #1

## Example

```go
cond := &tabwrap.Condition{TabWidth: 4}
result := cond.ExpandTabFunc("abc\tdef", func(nSpaces int) string {
    return "→" + strings.Repeat(" ", nSpaces-1)
})
// result: "abc→def"
```

## Test plan

- [x] Existing `TestExpandTab` passes (validates `ExpandTab` delegation is correct)
- [x] New `TestExpandTabFunc` covers: arrow marker, identity with spaces, tab at start, multiple tabs, newlines

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tabwrap_test.go [hunk 1]
```diff
@@ -96,6 +96,70 @@ func TestExpandTab(t *testing.T) {
...
+		want := "→···hi"
+		if got != want {
+			t.Errorf("ExpandTabFunc(%q) = %q, want %q", "\thi", got, want)
+		}
+	})
+
+	t.Run("multiple tabs", func(t *testing.T) {
+		t.Parallel()
+		got := c.ExpandTabFunc("a\tb\t", func(nSpaces int) string {
+			return "[" + strings.Repeat("-", nSpaces-2) + "]"
```
[the remark this answers is not in the document]
Comment (apstndb) [reply tabwrap_test.go:144]: Good catch\! Although this specific test input always yields nSpaces >= 3, the callback was indeed fragile. Fixed in 267c85f by adding a guard for nSpaces < 2.

File: tabwrap.go [hunk 2]
```diff
@@ -82,6 +82,17 @@ func (c *Condition) StringWidth(s string) int {
...
+	})
+}
+
+// ExpandTabFunc replaces every tab by calling fn with the number of spaces
+// the tab would normally expand to (based on the current column and tab width).
+// The column advances by nSpaces regardless of what fn returns, so the caller
+// is responsible for returning a string whose display width equals nSpaces if
+// alignment matters. Columns reset at each newline.
+func (c *Condition) ExpandTabFunc(s string, fn func(nSpaces int) string) string {
 	opts := c.options()
```
[the remark this answers is not in the document]
Comment (apstndb) [reply tabwrap.go:98]: Rather than adding a nil guard, we added a test that verifies nil fn panics (cf79216). This is consistent with Go standard library conventions — functions like `slices.SortFunc` and `sort.Slice` do not guard against nil function arguments. A silent fallback could mask bugs at the call site.

File: tabwrap_test.go [hunk 3]
```diff
@@ -96,6 +96,73 @@ func TestExpandTab(t *testing.T) {
...
+	t.Run("with newline", func(t *testing.T) {
+		t.Parallel()
+		got := c.ExpandTabFunc("ab\t\ncd\t", func(nSpaces int) string {
+			return "→" + strings.Repeat(" ", nSpaces-1)
+		})
+		want := "ab→ \ncd→ "
+		if got != want {
+			t.Errorf("ExpandTabFunc = %q, want %q", got, want)
+		}
+	})
```
[the remark this answers is not in the document]
Comment (apstndb) [reply tabwrap_test.go:163]: Added a panic test instead (cf79216). This documents the expected behavior explicitly: nil fn is a programming error and should panic, not silently fall back.
